src/test/java/org/sonatype/nexus/ci/iq/ClassFilterLoggingTestTrait.groovy

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2016-present Sonatype, Inc. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package org.sonatype.nexus.ci.iq
+
+import java.util.logging.Level
+import java.util.logging.LogRecord
+
+import org.jenkinsci.plugins.workflow.cps.CpsFlowExecution
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.jvnet.hudson.test.LoggerRule
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+/**
+ * Add this trait to any test that should have an extra verification that no JEP-200 marshalling errors have occurred.
+ * See INT-2407 for additional information and INT-418 for original work.
+ */
+trait ClassFilterLoggingTestTrait
+{
+  Logger logger = LoggerFactory.getLogger(ClassFilterLoggingTestTrait)
+
+  @Rule
+  public LoggerRule loggerRule = new LoggerRule()
+
+  def helpfulMessage = '''A possible JEP-200 marshalling error has occurred. It is likely that additional classes 
+will need to be added to the 'hudson.remoting.ClassFilter' file. See INT-2407 for additional info'''
+
+  @Before
+  def setupLoggingTest() {
+    loggerRule.capture(10).record(CpsFlowExecution, Level.WARNING)
+  }
+
+  @After
+  def runLoggingTest() {
+    for (LogRecord logRecord : loggerRule.records) {
```

Comment:
`loggerRule.records.findAll{ it.thrown }.each{ record -> assertNoClassFilterErrors(record.thrown) }`
:)

Reply:
Thanks [Kaptain](http://www.kellyrob99.com/blog/)! :smile: b97f7b4